v3_async_wip/v3_async_wip/edge_hsm.py

```diff
@@ -47,7 +47,8 @@ def __init__(
         self.generation_id = generation_id
         self.workload_uri = _format_socket_uri(workload_uri)
 
-    def get_certificate(self) -> str:
+    # TODO: Use async http to make use of this being a coroutine
```

Comment:
>   # TODO: Use async http to make use of this being a coroutine [](http://example.com/codeflow?start=1&length=63)

It might be an extra hassle because of the unix sockets code we use for edge.

Reply:
We'd have to approach it differently, I don't think the monkeypatch would work (although it might). But manually using a unixsocket Session object isn't a big hassle IMO. That said, to make sure this doesn't break, it's probably a good idea to wait to try and change this until the E2E tests are hooked up.